### PR TITLE
Imp/add pod annotations

### PR DESCRIPTION
**Pull request by vova3379**
add pod annotation to telegraf-operator to allow inject vault secrets like below 
```
podAnnotations:
  vault.hashicorp.com/agent-inject: "true"
```

### CODE REVIEW

**Comment by korjek** [comment]
> add pod annotation to telegraf-operator to allow inject vault secrets like below
> 
> ```
> podAnnotations:
>   vault.hashicorp.com/agent-inject: "true"
>   vault.hashicorp.com/role: "monitoring-telegraf"
>   vault.hashicorp.com/inject-status: "injected"
>   vault.hashicorp.com/agent-inject-secret-environment: "secret/ops/monitoring/telegraf"
> ```
I think if we going to create upstream PR, description should be more general

File: charts/telegraf-operator/Chart.yaml [hunk 1]
```diff
@@ -31,8 +31,8 @@ type: application
 
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
-version: 1.2.0
+version: 1.2.1
 
 # This is the version number of the application being deployed. This version number should be
 # incremented each time you make changes to the application.
-appVersion: v1.2.0
+appVersion: v1.2.1
```
Comment (mdelaney) [inline charts/telegraf-operator/Chart.yaml]: this is not targeting a different version of Telgeraf itself. I don't think this number should be bumped

File: charts/telegraf-operator/Chart.yaml [hunk 2]
```diff
@@ -31,8 +31,8 @@ type: application
 
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
-version: 1.2.0
+version: 1.2.1
```
Comment (mdelaney) [inline charts/telegraf-operator/Chart.yaml:34]: this is a feature add, bump minor version

File: charts/telegraf-operator/templates/deployment.yaml [hunk 3]
```diff
@@ -14,10 +14,13 @@ spec:
     metadata:
       labels:
         {{- include "telegraf-operator.selectorLabels" . | nindent 8 }}
-{{- if eq .Values.certManager.enable false }}
       annotations:
+{{- if eq .Values.certManager.enable false }}
         checksum/config: {{ include (print $.Template.BasePath "/tls.yml") . | sha256sum }}
 {{- end }}
+        {{- if .Values.podAnnotations }}
```
Comment (mdelaney) [inline charts/telegraf-operator/templates/deployment.yaml:21]: Add `podAnnotations` to `values.yaml`

Also, seems like it's more idiomatic to use `{{- with .Values.podAnnotations }}` similar to below

**Review by mdelaney — CHANGES_REQUESTED** [review]

File: charts/telegraf-operator/Chart.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (vova3379) [reply charts/telegraf-operator/Chart.yaml]: Already revert this change, checking which docker image chart load

Comment (vova3379) [reply charts/telegraf-operator/Chart.yaml]: Already revert this change, checking which docker image chart load

File: charts/telegraf-operator/values.yaml [hunk 4]
```diff
@@ -45,4 +45,5 @@ resources:
 nodeSelector: {}
 tolerations: []
 affinity: {}
+podAnnotations: []
```
Comment (korjek) [inline charts/telegraf-operator/values.yaml:48]: it's a dict

**Pull request closed without merging**